tests/unit/challenges/test_views.py

```diff
@@ -310,6 +310,39 @@ def test_remove_participant_team(self):
         self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
 
 
+class DeregisterParticipantTeamTest(BaseAPITestClass):
+    def setUp(self):
+        super(DeregisterParticipantTeamTest, self).setUp()
+
+        self.participant = Participant.objects.create(
+            user=self.user,
+            status=Participant.ACCEPTED,
+            team=self.participant_team,
+        )
+
+        self.challenge.participant_teams.add(self.participant_team)
+
+    def test_deregister_participant_team(self):
+        self.url = reverse_lazy(
+            "challenges:deregister_participant_team_from_challenge",
+            kwargs={"challenge_pk": self.challenge.pk},
+        )
+
+        response = self.client.post(self.url, {})
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data["success"], "Successfully deregistered!")
+
```

Comment:
Is it not possible to add a test for the case when there are existing submissions?

Reply:
I checked and yes we can do it too , i will add it